File: ejercicio5.py
'''
Created on Nov 28, 2022

@author: estudiante
'''

'''
Diseñar una función llamada palíndromo que tiene una cadena de
caracteres como parámetro de entrada, y devuelve Verdadero si 
es un palíndromo o Falso en otros casos. Una palabra es un
palíndromo si puede leerse de izquierda a derecha o de derecha 
a izquierda, ignorando los blancos. Por ejemplo: "anilina" o
"Dabale arroz a la zorra el abad" Para simplificar el problema, 
se puede suponer que se usan caracteres simples, es decir, 
sin tildes ni diresis.
'''

cadena = "Dabale arroz a la zorra el abad".lower()
cadena = cadena.replace(" ","")
cadena2 = cadena[::-1]
def palindromo(cadena):
    esPalindromo = False
    if cadena == cadena2: 
        esPalindromo = True 
    return esPalindromo
print(palindromo(cadena))

File: Ejercicio5.java
package boletin3.ejercicios;

import java.util.Scanner;

public class Ejercicio5 {

	public static void main(String[] args) {
		/*
		Diseña una función llamada esPalindromo que reciba una cadena de caracteres y
		determine si constituye un palíndromo o no. Una palabra es un palíndromo si puede
		leerse del mismo modo de izquierda a derecha que de derecha a izquierda. Obvia
		los espacios en blanco y caracteres separadores, así como tildes, etc.
		Ejemplos de palíndromos: ‘Ligar es ser ágil’, ‘Somos o no somos’.
		*/
		
		Scanner sc = new Scanner(System.in);
		System.out.println("Escriba una cadena: ");
		String cadena = sc.nextLine().toLowerCase();
		System.out.print(esPalindromo(cadena));
		

	}
	public static boolean esPalindromo(String cadena) {
		int i = 0;
		int contador = 0;
		boolean palindromo = false;
		String cadena_nueva = cadena.replace(" ","").replace("á","a").replace("é","e").replace("í","i").replace("ó","o").replace("ú","u");
		for (int j = cadena_nueva.length()-1; j>=0;) {
			while (i<cadena_nueva.length()) {
				if (cadena_nueva.charAt(i)==cadena_nueva.charAt(j)) {
					contador++;
				}
				i++;	
				j--;
			}
		}
		if (contador==cadena_nueva.length()) {
			palindromo = true;
		}
		return palindromo;
	}

}
